pkg/broker/broker.go

```diff
@@ -101,16 +101,15 @@ func (a AnsibleBroker) getServiceInstance(instanceUUID uuid.UUID) (*apb.ServiceI
 
 }
 func (a AnsibleBroker) Login() error {
-	a.log.Debug("Retrieving serviceaccount token")
-	token, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	clientConfig, err := k8srestclient.InClusterConfig()
 	if err != nil {
-		a.log.Debug("Error reading serviceaccount token")
+		a.log.Warning("Failed to create a restclient.InClusterConfig: %v.", err)
 		return err
 	}
 
-	return apb.OcLogin(a.log, "https://kubernetes.default",
-		"--certificate-authority", "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
-		"--token", string(token),
+	return apb.OcLogin(a.log, clientConfig.Host,
+		"--certificate-authority", clientConfig.CAFile,
```

Comment:
@rthallisey please look again, this OcLogin is actually opposite of what you said:

https://github.com/jwmatthews/ansible-service-broker/blob/34f23d72cef22bef8dbf221df41b05a5b6c838b2/pkg/broker/broker.go#L80

OcLogin for this case is used when running with InCluster.
It is intended to get a ~/.kube/config so that the oc create commands for role bindings and the like for dynamic service account creation will function.

This call is __not__ called from RefreshLoginToken, this is called from NewAnsibleBroker()